spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestDelete.java

```diff
@@ -749,6 +772,46 @@ public void testDeleteRefreshesRelationCache() throws NoSuchTableException {
     spark.sql("UNCACHE TABLE tmp");
   }
 
+  @Test
+  public void testDeleteWithMultipleSpecs() {
+    createAndInitTable("id INT, dep STRING, category STRING");
+
+    // write an unpartitioned file
+    append(tableName, "{ \"id\": 1, \"dep\": \"hr\", \"category\": \"c1\"}");
+
+    // write a file partitioned by dep
+    sql("ALTER TABLE %s ADD PARTITION FIELD dep", tableName);
+    append(tableName,
+        "{ \"id\": 1, \"dep\": \"hr\", \"category\": \"c1\" }\n" +
+            "{ \"id\": 2, \"dep\": \"hr\", \"category\": \"c1\" }");
+
+    // write a file partitioned by dep and category
+    sql("ALTER TABLE %s ADD PARTITION FIELD category", tableName);
+    append(tableName, "{ \"id\": 1, \"dep\": \"hr\", \"category\": \"c1\"}");
```

Comment:
This test would be a lot more readable if the rows were unique rather than reusing id=1 in several of them.

Reply:
Maybe use evens vs odds to delete.

Reply:
The idea of this test is to create files with `id = 1` in a number of specs and then issue a single delete that would remove all matching records and write delete files that belong to multiple specs. That's why multiple files contain the same record.

Reply:
My point is just that it doesn't need to be the same record in all places. That's why I suggested odds vs evens. If you use a different category and ID, then delete all the odd ones you would end up with the same test without the reader wondering why they all have the same ID.